spec/acceptance/management_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper_acceptance'
+
+describe 'rancher::management' do
+  context 'puppet apply' do
+    it 'is expected to apply and be idempotent' do
+      pp = <<-EOS
+class { 'rancher::management':
+  rancher_manager_port => '8080',
+}
+EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      apply_manifest(pp, :catch_changes => true)
+    end
+
+    context 'is expected to be running on the default port' do
+      before :each do
+        sleep 5 # Give it time to start.
```

Comment:
How about something along these lines:

```ruby
  #This will fail if your laptop (and therefor the Vagrant vm) is not running on AC power
  describe package('puppet') do
    it {
      sleep(240) #Wait for the task to execute
      should be_installed.with_version('3.5.1')
    }
  end
```

From Liam Bennett's blog [here](http://tech.opentable.co.uk/blog/2014/09/01/testing-puppet-with-beaker-pt-dot-2-the-windows-story/).